src/Microsoft.DotNet.ProjectJsonMigration/ProjectMigrator.cs

```diff
@@ -33,14 +35,51 @@ public ProjectMigrator(IMigrationRule ruleSet)
             _ruleSet = ruleSet;
         }
 
-        public void Migrate(MigrationSettings migrationSettings)
+        public void Migrate(MigrationSettings settings, bool skipProjectReferences = false)
         {
-            var migrationRuleInputs = ComputeMigrationRuleInputs(migrationSettings);
-            VerifyInputs(migrationRuleInputs, migrationSettings);
+            if (settings == null)
+            {
+                throw new ArgumentNullException();
+            }
 
-            SetupOutputDirectory(migrationSettings.ProjectDirectory, migrationSettings.OutputDirectory);
+            Queue<MigrationSettings> settingsQueue = new Queue<MigrationSettings>();
+            settingsQueue.Enqueue(settings);
+
+            while(settingsQueue.Count() != 0)
```

Comment:
What do you think about factoring this into another class or method that returns the full lists of projects to go migrate?